src/game/server/entities/character.cpp

```diff
@@ -3160,22 +3160,39 @@ void CCharacter::Snap(int SnappingClient)
 		else if(GetClass() == PLAYERCLASS_HERO && g_Config.m_InfHeroFlagIndicator && m_pHeroFlag) 
 		{
 			CHeroFlag *pFlag = m_pHeroFlag;
+
+			long inactiveTicks = m_pPlayer->m_LastActionTick+g_Config.m_InfHeroFlagIndicatorTime*Server()->TickSpeed();
 			
 			// Guide hero to flag
-			if(pFlag->GetCoolDown() <= 0)
+			if(pFlag->GetCoolDown() <= 0 && Server()->Tick() > inactiveTicks)
 			{
-				CNetObj_Projectile *pObj = static_cast<CNetObj_Projectile *>(Server()->SnapNewItem(NETOBJTYPE_PROJECTILE, m_CursorID, sizeof(CNetObj_Projectile)));
+				CNetObj_Laser *pObj = static_cast<CNetObj_Laser *>(Server()->SnapNewItem(NETOBJTYPE_LASER, m_CursorID, sizeof(CNetObj_Laser)));
 				if(!pObj)
 					return;
 
 				float Angle = atan2f(pFlag->m_Pos.y-m_Pos.y, pFlag->m_Pos.x-m_Pos.x);
 				vec2 Indicator = m_Pos + vec2(cos(Angle), sin(Angle)) * 84.0f; 
+				vec2 IndicatorM = m_Pos - vec2(cos(Angle), sin(Angle)) * 84.0f; 
 
 				pObj->m_X = (int)Indicator.x;
 				pObj->m_Y = (int)Indicator.y;
-				pObj->m_VelX = 0;
-				pObj->m_VelY = 0;
+				pObj->m_FromX = pObj->m_X;
+				pObj->m_FromY = pObj->m_Y;
 				pObj->m_StartTick = Server()->Tick();
+
+				// display laser beam for 0.5 seconds
+				if (g_Config.m_InfHeroFlagIndicatorTime > 0 && Server()->Tick()-inactiveTicks < Server()->TickSpeed()*0.5) 
```

Comment:
> `if (g_Config.m_InfHeroFlagIndicatorTime > 0 ...)`

Maybe 0 could mean that indicator should always be displayed?

Reply:
Yes that is the case, but the laser line will not be shown. Only the laser dot.